drivers/CoreEventQueue/include/CoreEventQueue.h

```diff
@@ -18,19 +18,23 @@ class CoreEventQueue : public interface::EventQueue
 
 	void dispatch_forever() final;
 
-	void call(auto f, auto... params) { _event_queue.call(f, params...); }
+	void call(auto f, auto... params) { _event_id = _event_queue.call(f, params...); }
```

Comment:
here, the `_event_id` is not really needed and could even introduce bugs.

`_event_queue.call` will dispatch the event as soon as it's called .

per mbed documentation, when the event is dispatched, the id is not valid anymore and

> It is not safe to call cancel after an event has already been dispatched.
> id must be valid i.e. event must have not finished executing.

https://os.mbed.com/docs/mbed-os/v6.15/mbed-os-api-doxy/classevents_1_1_event_queue.html#a9ccb4f1222149f37161430fa2f1974e4

so calling `call` and then `cancel` might break the program.

cancel is only really needed for `call_every` or `call_in`